### PR TITLE
Cards

### PR DESCRIPTION
**Pull request by pascalweber85**
ready for a review:)

### CODE REVIEW

File: src/components/Card.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  )
+}
+
+const Wrapper = styled.section`
+  padding: 20px;
+  border-radius: 8px;
+  box-shadow: 0 8px 16px #0006;
+  display: flex;
+  gap: 10px;
+  justify-items: end;
```
Comment (Morten-Moeller) [inline src/components/Card.js]: justify-content is used for display flex.

File: src/components/Card.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  img {
+    justify-self: start;
+    width: 150px;
+  }
+
+  h2 {
+    font-size: 18px;
+    word-wrap: break-word;
+    width: 9em;
```
Comment (Morten-Moeller) [inline src/components/Card.js]: consider using ch instead of em for the width

File: src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    90deg,
+    rgb(238, 171, 48),
+    orange,
+    rgba(255, 81, 0, 0.719),
+    orange,
+    rgb(241, 165, 23)
+  );
+  padding: 20px;
+  color: whitesmoke;
+  font-family: Arial, Helvetica, sans-serif;
```
Comment (Morten-Moeller) [inline src/components/Header.js]: Didn't you set the font-family inside GlobalStyles?

File: src/pages/locationPage.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+export default function LocationPage({ title }) {
+  return (
+    <Grid>
+      <Header>
+        <h2>Best Locations</h2>
+      </Header>
+      <div>
+        {location.map(location => {
+          const { id, attributes } = location
```
Comment (Morten-Moeller) [inline src/pages/locationPage.js]: you could de-structure in line 18 and you can on top de-structure attributes too

**Review by Morten-Moeller — COMMENTED** [review]
Good job, added some small suggestions. 👾
Happy coding

File: src/GlobalStyle.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { createGlobalStyle } from 'styled-components'
+
+export default createGlobalStyle`
+ * {
+   box-sizing: border-box
```
Comment (macfrei) [inline src/GlobalStyle.js]:
```suggestion
   box-sizing: border-box;
```

File: src/components/Card.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import PropTypes from 'prop-types'
+
+Card.propTypes = {
+  image: PropTypes.node,
+  title: PropTypes.string,
+  ratingimage: PropTypes.node,
+  text: PropTypes.string,
+}
+
+export default function Card({ image, title, ratingimage, text }) {
```
Comment (macfrei) [inline src/components/Card.js]: To keep it in camelCase: `ratingImage`
```suggestion
export default function Card({ image, title, ratingImage, text }) {
```

File: src/components/Card.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  return (
+    <Wrapper>
+      <img src={image} alt="" />
+      <div>
+        <h2>{title}</h2>
+        <img src={ratingimage} alt="" />
```
Comment (macfrei) [inline src/components/Card.js]: Does `ratingImage` contain information about how the location is rated? Is there a rating displayed? If so, then the information should be provided in the `alt` attribute.

File: src/components/Card.spec.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { render, screen } from '@testing-library/react'
+import Card from './Card'
+
+describe('Card', () => {
+  it('renders name of card', () => {
+    render(<Card title="Völkerschlachtdenkmal" />)
+    expect(screen.getByText('Völkerschlachtdenkmal')).toBeInTheDocument()
+  })
```
Comment (macfrei) [inline src/components/Card.spec.js]: I would test all props and with the right role.
```suggestion
  it('renders name of card', () => {
    render(<Card title="Völkerschlachtdenkmal" text="Lorem Ipsum"  image="path/to/image" ratingImage="path/to/Image" />)
    expect(screen.getByRole('heading', {name: 'Völkerschlachtdenkmal'})).toBeInTheDocument()
    ....
  })
```

File: src/components/Card.spec.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+  it('renders text of card', () => {
+    render(<Card text="heading" />)
+    expect(screen.getByRole('heading')).toBeInTheDocument()
+  })
```
Comment (macfrei) [inline src/components/Card.spec.js]: This does _not_ test the text! When you query be `getByRole('heading')` you get the `h2` element of your card.

File: src/components/Card.stories.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+const Template = args => <Card {...args} />
+
+export const Default = Template.bind({})
+Default.args = {
+  image: imageFile,
+  title: 'Völkerschlachtdenkmal',
+  ratingimage: imageFile2,
+  text: 'Sehenswürdigkeit',
```
Comment (macfrei) [inline src/components/Card.stories.js]: These should be tested as well. Like I commented above.

File: src/components/Header.spec.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import { render, screen } from '@testing-library/react'
+import Header from './Header'
+
+describe('Header', () => {
+  it('render a children', () => {
+    const { container } = render(<Header>Hello</Header>)
+    expect(container.firstChild).toHaveTextContent('Hello')
+    expect(screen.getByRole('heading')).toBeInTheDocument()
```
Comment (macfrei) [inline src/components/Header.spec.js:8]: One of them is efficient enough.

File: src/data.json [hunk 12]
```diff
@@ -0,0 +1,192 @@
+[
+  {
+    "type": "card",
```
Comment (macfrei) [inline src/data.json]: What is `"type": "card"` for?

File: src/pages/locationPage.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+          return (
+            <Card
+              key={id}
+              image={attributes.main_image_path}
+              title={attributes.title}
+              ratingimage={attributes.rating_image_path}
+              text={attributes.text}
+            />
+          )
+        })}
```
Comment (macfrei) [inline src/pages/locationPage.js]: Could be shortened to this:
```suggestion
        {location.map(({id, {main_image_path, rating_image_path, title, text}}) => (
            <Card
              key={id}
              image={main_image_path}
              title={title}
              ratingimage={rating_image_path}
              text={text}
            />
          )
}
```

File: src/pages/locationPage.spec.js [hunk 14]
```diff
@@ -0,0 +1,9 @@
+import { render, screen } from '@testing-library/react'
+import LocationPage from './LocationPage'
+
+describe('LocationPage', () => {
+  it('renders', () => {
+    render(<LocationPage />)
+    expect(screen.getByText('LocationPage')).toBeInTheDocument()
+  })
+})
```
Comment (macfrei) [inline src/pages/locationPage.spec.js]: Looks like you forgot to test it!

File: src/pages/locationPage.js [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+      </div>
+    </Grid>
+  )
+}
+
+const Grid = styled.section`
+  padding: 10px;
+`
```
Comment (macfrei) [inline src/pages/locationPage.js]: If you call it `Grid` I would expect that there is a grid but here is none.

File: src/pages/locationPage.js [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+import PropTypes from 'prop-types'
+import Card from '../components/Card'
+import location from '../data.json'
+import Header from '../components/Header'
+
+LocationPage.propTypes = {
+  title: PropTypes.string,
+}
+
+export default function LocationPage({ title }) {
```
Comment (macfrei) [inline src/pages/locationPage.js]: You don't use this title anywhere....
```suggestion
export default function LocationPage() {
```

File: src/pages/locationPage.stories.js [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+export default {
+  title: 'LocationPage',
+  component: LocationPage,
+}
+
+const Template = args => <LocationPage {...args} />
+
+export const Default = Template.bind({})
+Default.args = {
+  title: 'All Locations',
```
Comment (macfrei) [inline src/pages/locationPage.stories.js]: This doesn't work, because you don't use it anywhere.

**Review by macfrei — CHANGES_REQUESTED** [review]
Well done, added a few comments you might want to consider! 🚀

**Pull request merged**